src/Interval.php

```diff
@@ -0,0 +1,98 @@
+<?php declare(strict_types=1);
+
+namespace Amp;
+
+use Revolt\EventLoop;
+
+/**
+ * This object invokes the given callback within a new coroutine every $interval seconds until either the
+ * {@see self::disable()} method is called or the object is destroyed.
+ */
+final class Interval
+{
+    private readonly string $watcher;
+
+    /**
+     * @param float $interval Invoke the function every $interval seconds.
+     * @param \Closure():void $closure Use {@see weakClosure()} to avoid a circular reference if storing this object
+     *      as a property of another object.
+     * @param bool $reference If false, unreference the underlying watcher.
+     */
+    public function __construct(float $interval, \Closure $closure, bool $reference = true)
+    {
+        $this->watcher = EventLoop::repeat($interval, $closure);
+
+        if (!$reference) {
+            EventLoop::unreference($this->watcher);
+        }
+    }
+
+    public function __destruct()
+    {
+        EventLoop::cancel($this->watcher);
+    }
```

Comment:
Shouldn't there also be a manual `cancel` method?

Reply:
I think it makes more sense for the object to represent a valid repeating timer. If we allow external cancellation, then the object needs to expose an `isCancelled()` method so the state isn't hidden.